### DISCUSSION TITLE
Elastalert rules directory

### DISCUSSION THREAD

**Discussion by launchdaemon** [Security-Onion-Solutions/securityonion#3316] [Unsupported Versions] [answered]
Hi,

I have a standalone installation of Security Onion on version 2.3.30, installed on Ubuntu 18.04. I've been trying to configure elastalerts and read through both the security onion and elastalert documentation.

I had a query regarding the directory rules should be in as there seem to be some differences and I wasn't sure if I was just misunderstanding or documentation had been updated at different times.

The Security Onion documentation [here](https://docs.securityonion.net/en/latest/elastalert.html#email---internal), says:

> ElastAlert rules are stored in `/opt/so/conf/elastalert/`.

That doesn't contain a `rules` directory and if you put a `.yaml` file there it is not found by `so-elastalert-test`.

`elastalert_config.yaml` in that directory defines:

> `rules_folder: /opt/elastalert/rules/`

however putting any rules in there also don't seem to be found when running `so-elastalert-test`.

Running `so-elastalert-test` specifies that rules should be stored in `/opt/so/rules/elastalert/` and when you run the tool it does pick them up but I wasn't sure if the system will find them there if it is defined differently in config yaml?

Regarding creating the rules with `so-elastalert-create` should I start a different discussion here or on the repo for that?  I wasn't sure if Security Onion maintained it's own version or modified it at all?

**Comment by launchdaemon** [comment]
When you run `so-elastalert-restart` it checks the state of `/opt/so/rules/elastalert` so assume that is the correct one?

**Reply by launchdaemon** [reply]
I don't understand where it is getting this from though as the output from that command also says it is

> Executing state file.managed for `/opt/so/conf/elastalert/elastalert_config.yaml`

which defines it differently as mentioned.

And the actual log file is different to that mentioned in the config.  If I `tail -f /opt/so/log/elastalert/elastalert.log` I can see it running my rule in `/opt/so/rules/elastalert/`.

**Reply by 8inaryMata1eao** [reply]
Noticed the same issue and surprised no one else is having the issue or asking about it.  Coped the rules from playbook and wrote my own then ...soup blew them away.  I also noticed that the instructions on how to setup the email alerting isn't 100% clear. Do we add each email alert to the playbook rule in the web UI or manually modify the text in the /opt/so/rules/elastalert/playbook/.yaml or do we modify the /opt/so/conf/elastalert/elastalert_config.yaml?

**Comment by jertel** [comment, marked as answer]
On the host, the correct location for rules is: `/opt/so/rules/elastalert`.
Inside the so-elastalert container, this directory is mapped to `/opt/elastalert/rules`.

Place new rules inside the `/opt/so/rules/elastalert` on the host and both so-elastalert-test and restarting the so-elastalert container will find them.

Yes, SMTP documentation for email alerting is not yet fully updated from 16.04 and needs work. Any contribution you'd like to provide will be appreciated.
